remove buttons pull request

siddharthsingh opened the pull request:
Remove get centers and get districts button

Commit:
Remove buttons v1

Commit:
Remove buttons v2

Commit:
add disctrict info

Commit:
add disctrict info

lihas commented on index.js:
```diff
@@ -264,7 +269,7 @@ window.mobileCheck = function() {
                 })
             }
-            centreData = Object.assign({}, {name:centre["name"]}, daysData);
+            centreData = Object.assign({}, {name:centre["name"], name_district:centre["name"]+" ("+centre["district"]+")"}, daysData);
```
Rather than constructing string to render here, just add required data, and use "render:" of data tables. Search this source for "render:" for sample usage.

lihas replied:
```suggestion
            centreData = Object.assign({}, {name:{name:centre["name"], district:centre["district"]}, daysData);
```

Here instead of creating string we are just storing info which we will use in "render:" for constructing the string

lihas commented on index.js:
```diff
@@ -386,13 +391,15 @@ window.mobileCheck = function() {
     function CreateTable(bCallAlarm = false /*Show notification, and sound alarm*/, bAutoScroll = true /*Auto scroll down to results table*/, bShowScrollNotif = true/*Show notification that you may have to scroll*/)
     {
-        let tableColumns = [{data: 'name', title: 'Centre name'}];
+        let tableColumns = [{data: 'name_district', title: 'Centre name'}];
```
```suggestion
        let tableColumns = [{data: 'name', title: 'Centre name'}, render: function(data,type){return data.name + '(' +data.district + ')';}];
```
The data that you stored above - name and district will be available here in this functions data object. Whatever this render functions returns will be rendered in the corresponding cell of the table. This function can return string, html string, etc.

eg. if you want to make district in bold you can change this code further to:
```suggestion
        let tableColumns = [{data: 'name', title: 'Centre name'}, render: function(data,type){return data.name + '(<b>' +data.district + '</b>)';}];
```

lihas commented on index.js:
```diff
@@ -578,7 +586,7 @@ window.mobileCheck = function() {
                     });
                     
-                    let cInfo = {name:cName, sessions:cSessions2};
+                    let cInfo = {name:cName, district:cDistrictName, sessions:cSessions2};
```
This is fine

lihas commented on index.js:
```diff
@@ -932,6 +941,7 @@ window.mobileCheck = function() {
             {
                 AddRemoveUrlParam(false, 'states', stateInt);
                 g_statesSelected.delete(stateInt);
+                GetDistricts();
```
Rather than using onAdd(), or onRemove() you can use onShow(), onHide(). This will result in lesser number of API calls.
onAdd()/remove gets called every time a state/district is added. onShow()/hide get called only when user has selected/removed all states he/she wants to and the dropdown closes.
This will not affect functionality only perf.
Just a suggestion. Final call you can take.
We can also make the perf improvement in future and check in the code as is right now.

lihas commented on index.md:
```diff
@@ -115,22 +115,18 @@ title: Covid Shots
 </select>
 
-
+<br>
```
Remove this<br />. Vertical space between the two dropdowns without this seems enough. One reason to remove getdistrict and centre buttons was to reduce vertical space so that the site looks better on mobile

lihas commented on index.js:
```diff
@@ -386,13 +391,15 @@ window.mobileCheck = function() {
     {
-        let tableColumns = [{data: 'name', title: 'Centre name'}];
+        let tableColumns = [{data: 'name_district', title: 'Centre name'}];
+        // tableColumns.push({data: 'name_district', title: 'District'});
```
Remove commented code

lihas reviewed (changes requested):
Add more description to pull request. This change doesn't just remove buttons it also adds district info to table.
Make the changes for "render:" method I have suggested. That will make it easier to make changes to table UI in future.
Remove the extra "<br />" between the two dropdowns.

Commit:
changes based on pull request comments

siddharthsingh commented:
Changes:
1. Removed get district and get centers button and modified their click events to fire when drop-down hides
2. Added district name in the table.
3. The center name is still not bold as written in javascript -> Need to work on that

Commit:
changes based on pull request comments

lihas reviewed (approved):
Changes look good. will merge
